src/providers/ogr/qgsogrfeatureiterator.cpp

```diff
@@ -45,6 +45,10 @@ QgsOgrFeatureIterator::QgsOgrFeatureIterator( QgsOgrFeatureSource* source, bool
   mFeatureFetched = false;
 
   mConn = QgsOgrConnPool::instance()->acquireConnection( mSource->mFilePath );
+  if ( mConn->ds == nullptr )
```

Comment:
OK, I personnaly find == nullptr more explicit, but tastes may vary. Is https://www.qgis.org/en/site/getinvolved/development/qgisdevelopersguide/codingstandards.html the more up-to-date doc to refer to ?